IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -117,35 +120,25 @@ - (void)updateWithApiId:(NSInteger)apiId
 {
     if (errorString)
     {
-        dispatch_barrier_async(self.synchronizationQueue, ^{
-            _errorsInfo = [_errorsInfo count] ? _errorsInfo : [NSMutableArray new];
-            
-            __auto_type errorInfo = [MSIDRequestTelemetryErrorInfo new];
-            errorInfo.apiId = apiId;
-            errorInfo.error = errorString;
-            errorInfo.correlationId = context.correlationId;
-            
-            [_errorsInfo addObject:errorInfo];
-        });
+        __auto_type errorInfo = [MSIDRequestTelemetryErrorInfo new];
+        errorInfo.apiId = apiId;
+        errorInfo.error = errorString;
+        errorInfo.correlationId = context.correlationId;
+        [self addErrorInfo:errorInfo];
     }
     else
     {
-        dispatch_barrier_async(self.synchronizationQueue, ^{
-            self.silentSuccessfulCount = 0;
-            _errorsInfo = nil;
-        });
+        [self resetSuccessfulCount];
+        [self resetErrorInfo];
```

Comment:
nit: just a small change. Can we move the reset logic to one method called reset something like 
- (void) reset {
dispatch_barrier_async(_synchronizationQueue, ^{
       _silentSuccessfulCount = 0
        _errorsInfo = nil;
        [self saveTelemetryToDisk];
    });
} This will be more performant as we will need only one dispatch_barrier instead of two.

Reply:
Definitely, good idea, I missed that.